### PR TITLE
move PCI capability testing to Python

### PR DESCRIPTION
**Pull request by jlevon**

### CODE REVIEW

**Review by swapnili — COMMENTED** [review]
I haven't looked into detail. It looks good. Just some minor comments.

File: test/py/libvfio_user.py [hunk 1]
```diff
@@ -40,7 +40,32 @@
 import struct
 import syslog
 
-SOCK_PATH = b"/tmp/vfio-user.sock.%d" % os.getpid()
+# from linux/pci_regs.h
```
[the remark this answers is not in the document]
Comment (jlevon) [reply test/py/libvfio_user.py:43]: done

File: test/py/libvfio_user.py [hunk 2]
```diff
@@ -130,14 +151,21 @@
 lib.vfu_attach_ctx.argtypes = (c.c_void_p,)
 lib.vfu_run_ctx.argtypes = (c.c_void_p,)
 lib.vfu_destroy_ctx.argtypes = (c.c_void_p,)
-lib.vfu_setup_region.argtypes = (c.c_void_p, c.c_int, c.c_long, c.c_void_p,
+region_cb_t = c.CFUNCTYPE(c.c_int, c.c_void_p, c.POINTER(c.c_char), c.c_long,
```
[the remark this answers is not in the document]
Comment (jlevon) [reply test/py/libvfio_user.py:154]: done

File: test/py/libvfio_user.py [hunk 3]
```diff
@@ -130,14 +151,21 @@
...
+                          c.c_long, c.c_int)
+lib.vfu_setup_region.argtypes = (c.c_void_p, c.c_int, c.c_long, region_cb_t,
                                  c.c_int, c.c_void_p, c.c_int, c.c_int)
```
[the remark this answers is not in the document]
Comment (jlevon) [reply test/py/libvfio_user.py:157]: done, I think

File: test/py/libvfio_user.py [hunk 4]
```diff
@@ -187,25 +215,111 @@ class vfu_pci_hdr_t(c.Structure):
...
+    sock = connect_sock()
+
+    json = b'{ "capabilities": { "max_fds": 8 } }'
+    # struct vfio_user_version
+    payload = struct.pack("HH%dsc" % len(json), LIBVFIO_USER_MAJOR,
+                          LIBVFIO_USER_MINOR, json, b'\0')
+    hdr = vfio_user_header(VFIO_USER_VERSION, size=len(payload))
+    sock.send(hdr + payload)
+    vfu_attach_ctx(ctx, expect=0)
+    payload = get_reply(sock, expect=0)
```
[the remark this answers is not in the document]
Comment (jlevon) [reply test/py/libvfio_user.py:253]: Sorry, I'm not sure what you mean by this.

[the remark this answers is not in the document]
Comment (swapnili) [reply test/py/libvfio_user.py:254]: Sorry I was not clear. I mean to verify the json data(version, capabilities) received in the payload.

[the remark this answers is not in the document]
Comment (jlevon) [reply test/py/libvfio_user.py:254]: We already have coverage for that:

https://github.com/nutanix/libvfio-user/blob/master/test/py/test_negotiate.py#L151

**Review by swapnili — APPROVED** [review]

**Pull request merged**